### PR TITLE
A0-1221: Extract migration support to a new crate

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# Description

We move migration support code to a new crate, so that our other pallets (for now: `aleph`) could benefit from better storage migration checks (A0-1222).

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- I have made neccessary updates to the Infrastructure
- I have bumped `spec_version` and `transaction_version`

### CODE REVIEW

File: pallets/elections/Cargo.toml [hunk 1]
```diff
@@ -37,8 +38,12 @@ std = [
     "pallet-session/std",
     "pallet-staking/std",
     "pallet-authorship/std",
+    "pallets-support/std",
```
Comment (Marcin-Radecki) [inline pallets/elections/Cargo.toml:41]: Could you please bump `version`? Tbh not sure whether this is breaking change or not, but I guess `0.4.0` would to the job.

File: pallets/support/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "pallets-support"
```
Comment (Marcin-Radecki) [inline pallets/support/Cargo.toml:2]: Any particular reason to name it like this? Ie is it some substrate convention? I was thinking of anything `migration` in the name

**Review by Marcin-Radecki — APPROVED** [review]

File: pallets/support/Cargo.toml (hunk 2 above)
Comment (pmikolajczyk41) [reply pallets/support/Cargo.toml:2]: I'm expecting that more things may appear here, so in general: support (like frame-support)

File: pallets/elections/Cargo.toml (hunk 1 above)
Comment (pmikolajczyk41) [reply pallets/elections/Cargo.toml:41]: sure, done

**Review by fbielejec — APPROVED** [review]

**Pull request merged**